src/actions/signupActions.js

```diff
@@ -0,0 +1,55 @@
+import request from '../utils/request';
+import typeGenerator from './typeGenerator';
+
+export const signupType = typeGenerator('SIGNUP');
+
+/**
+ * This function returs an action with its type
+ * @param {String} type - auto genrated action type
+ * @param {Object} data - data to be pased to store
+ */
+const signupAction = (type, data) => ({ type, data });
+
+/**
+ * This action signs up a user and populates the store
+ * @param {Object} userData - This is the user data to be sent to the API
+*/
+export const signUpUser = userData => async (dispatch) => {
+  dispatch(signupAction(signupType.loading, true));
+  try {
+    const response = await request({
+      route: 'users',
+      method: 'post',
+      payload: userData
+    });
+    const { data: { data } } = response;
+    dispatch(signupAction(signupType.success, {
+      loading: false,
+      success: true,
+      token: data.token
```

Comment:
@davdwhyte87 You would have to save the token in session storage too. This allows us to apply the singleton principle when making a request with axios.

Reply:
Ok Alright. https://github.com/andela/apollo-ah-frontend/blob/feature%2F164198142%2Fsignup/src%2Factions%2FsignupActions.js/#L26